packages/eventstream-handler-node/src/EventSigningStream.ts

```diff
@@ -32,11 +34,12 @@ export class EventSigningStream extends Transform {
     this.priorSignature = options.priorSignature;
     this.eventStreamCodec = options.eventStreamCodec;
     this.messageSigner = options.messageSigner;
+    this.systemClockOffsetProvider = options.systemClockOffsetProvider;
   }
 
   async _transform(chunk: Uint8Array, encoding: string, callback: TransformCallback): Promise<void> {
     try {
-      const now = new Date();
+      const now = new Date(new Date().getTime() + (await this.systemClockOffsetProvider()));
```

Comment:
this uses `new Date().getTime()` instead of `Date.now()` due to a complex mock of the `Date` global in the unit tests.